src/client/automation/playback/click/index.js

```diff
@@ -225,12 +225,20 @@ export default class ClickAutomation {
         return cursor
             .buttonUp()
             .then(() => {
-                var prevElementForEvent = this.eventArgs.element;
-
-                this.eventArgs            = this._calculateEventArguments();
-                this.eventState.skipClick = prevElementForEvent !== this.eventArgs.element;
-
-                eventSimulator.mouseup(this.eventArgs.element, this.eventArgs.options);
+                var topElementEventArgs = this._calculateEventArguments();
+
+                if (this.eventArgs.element !== topElementEventArgs.element) {
+                    // NOTE: Mozilla Firefox always prevents click emulation
+                    // if the element under cursor has been changed.
+                    if (browserUtils.isFirefox)
+                        this.eventState.skipClick = true;
+                    else {
+                        this.eventState.skipClick = !this.eventArgs.element.contains(topElementEventArgs.element) &&
+                                                    !topElementEventArgs.element.contains(this.eventArgs.element);
```

Comment:
`!topElementEventArgs.element.contains(this.eventArgs.element)` - there is no scenario for this condition, my fault.